scripts/src/primazactl/cmd/options/options.py

```diff
@@ -0,0 +1,98 @@
+import yaml
+from primazactl.utils import logger
+from .tenant import Tenant
+from .cluster_environment import ClusterEnvironment
+from .defaults import defaults
+
+#  apiVersion: primaza.io/v1alpha1
+#  kind: Tenant
+#  name: alice
+#  kubeconfig: ~/.kube/config
+#  context: primaza
+#  version: latest
+#  manifest_directory: ./out/config
+#  clusterEnvironments:
+#    - name: onprem-cluster
+#      environment: dev
+#      targetCluster:
+#           kubeconfig: ~/.kube/config
+#           context: onprem
+#      applicationNamespaces:
+#           - name: alice-app
+#      serviceNamespaces:
+#           - name: alice-svc
+#    - name: aws-cluster
+#      environment: dev
+#      targetCluster:
+#           kubeconfig: ~/.kube/config-aws
+#           context: aws
+#      ["applicationNamespaces:"]
+#           - name: aws-app
+#       serviceNamespaces:
+#           - name: aws-svc
+
+API_VERSION: str = "apiVersion"
+KIND: str = "kind"
+
+
+class Options(object):
+
+    options: {} = None
+    options_empty: bool = True
+
+    def __init__(self, args):
+
+        logger.log_info(f"Options:{args.options_file}:")
+
+        if args.options_file:
+            with open(str(args.options_file), "r") as options_content:
+                load_options = yaml.safe_load(options_content)
+                logger.log_info(f"loaded options: {load_options}")
+
+            if API_VERSION in load_options and \
+                    load_options[API_VERSION] == defaults["apiVersion"]:
+                if KIND in load_options and load_options[KIND] \
+                        == defaults["kind"]:
+                    self.options = load_options
+                    self.options_empty = False
+                    logger.log_info(f"Option file content: {self.options}")
+                else:
+                    logger.log_warning("Invalid \'kind\' value in options "
+                                       "file, options file will be ignored")
+            else:
+                logger.log_warning("Invalid \'apiVersion\' value in options "
+                                   "file, options file will be ignored")
+        else:
+            logger.log_info("no options file")
+            self.options = {}
+
+    def get_options(self):
+        return self.options
+
+    def get_tenant(self):
+        return Tenant(self.options)
+
+    def get_cluster_environments(self, tenant):
+        cluster_environments = []
+        if "clusterEnvironments" in self.options:
+            for cluster_environment in self.options["clusterEnvironments"]:
+                cluster_environments.append(
+                    ClusterEnvironment(cluster_environment, tenant))
+        return cluster_environments
```

Comment:
this can be simplified as follows:
```suggestion
    def get_cluster_environments(self, tenant):
        return [ClusterEnvironment(ce, tenant) for ce in self.options.get("clusterEnvironments", [])]
```

Reply:
Personally I think that is less lines of code but more difficult to understand.

Reply:
Seems to me no changes have been pushed. I agree it's harder to understand. I'd propose to at least remove the if and use the `get("clusterEnvironments", [])` to remove one indentation level